core/src/views/radio_buttons.rs

```diff
@@ -0,0 +1,78 @@
+use std::cmp::min;
+use femtovg::{Paint, Path};
+use crate::{Canvas, Context, Event, Handle, MouseButton, Units, View, WindowEvent};
+use crate::style::PropGet;
+
+pub struct RadioButton {
+    on_select: Option<Box<dyn Fn(&mut Context)>>,
+}
+
+impl RadioButton {
+    pub fn new(cx: &mut Context, checked: bool) -> Handle<Self> {
+        Self { on_select: None }
+            .build2(cx, |_| {})
+            .checked(checked)
+    }
+}
+
+impl View for RadioButton {
+    fn element(&self) -> Option<String> {
+        Some("radiobutton".to_string())
+    }
+
+    fn event(&mut self, cx: &mut Context, event: &mut Event) {
+        if let Some(WindowEvent::MouseDown(MouseButton::Left)) = event.message.downcast() {
+            if let Some(f) = self.on_select.take() {
+                (f)(cx);
+                self.on_select = Some(f);
+            }
+        }
+    }
+
+    fn draw(&self, cx: &Context, canvas: &mut Canvas) {
+        let entity = cx.current;
+        let bounds = cx.cache.get_bounds(entity);
+        let border_width = if bounds.w < bounds.h { bounds.w } else { bounds.h } / 6.0;
```

Comment:
Where does this come from? Is this just what makes it look correct? I only ask because there is a border width style property already which can be specified by the user. My thinking is that it would be good for the user to be able to configure the border width as well as the size (radius?) of the inner circle.

Reply:
I just pushed a fix to use the border-width property. I initially didn't use it because I couldn't come up with a good way to make the inner dot stylable by the user, but I ended up just taking half of the leftover space after drawing the border and allocating it to the dot.